android/src/test/java/com/instabug/reactlibrary/RNInstabugReactnativeModuleTest.java

```diff
@@ -69,5 +81,492 @@ public Boolean answer(InvocationOnMock invocation) throws Throwable {
+            RNInstabugReactnativeModule mockModule = mock(RNInstabugReactnativeModule.class);
+            PowerMockito.doCallRealMethod().when(mockModule, "setPreInvocationHandler", Matchers.any());
+            PowerMockito.doAnswer(new Answer<Object>() {
+                @Override
+                public Object answer(InvocationOnMock invocation) {
+                    ((OnInvokeCallback) invocation.getArguments()[0]).onInvoke();
+                    return null;
+                }
+            }).when(BugReporting.class, "setOnInvokeCallback", Matchers.anyObject());
+            // when
+            mockModule.setPreInvocationHandler(null);
+            // then
+            verifyPrivate(mockModule, VerificationModeFactory.times(1))
+                    .invoke("sendEvent", any(), eq("IBGpreInvocationHandler"), eq(null));
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void given$setPostInvocationHandler_whenQuery_thenShouldSetNativeCallback() {
+        try {
+            // given
+            PowerMockito.mockStatic(BugReporting.class);
+            PowerMockito.mockStatic(Arguments.class);
+            RNInstabugReactnativeModule mockModule = mock(RNInstabugReactnativeModule.class);
+            PowerMockito.doCallRealMethod().when(mockModule, "setPostInvocationHandler", Matchers.any());
+            PowerMockito.when(Arguments.createMap()).thenReturn(new JavaOnlyMap());
+            PowerMockito.doAnswer(new Answer<Object>() {
+                @Override
+                public Object answer(InvocationOnMock invocation) {
+                    ((OnSdkDismissCallback) invocation.getArguments()[0])
+                            .call(OnSdkDismissCallback.DismissType.CANCEL, OnSdkDismissCallback.ReportType.BUG);
+                    return null;
+                }
+            }).when(BugReporting.class, "setOnDismissCallback", Matchers.anyObject());
+            // when
+            mockModule.setPostInvocationHandler(null);
+            // then
+            WritableMap params = new JavaOnlyMap();
+            params.putString("dismissType", OnSdkDismissedCallback.DismissType.CANCEL.toString());
+            params.putString("reportType", OnSdkDismissCallback.ReportType.BUG.toString());
+            verifyPrivate(mockModule, VerificationModeFactory.times(1))
+                    .invoke("sendEvent", any(), eq("IBGpostInvocationHandler"), eq(params));
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void givenArray$setReportTypes_whenQuery_thenShouldCallNativeApiWithEnumArgs() {
+        // given
+        PowerMockito.mockStatic(BugReporting.class);
+        final Map<String, Object> args = new HashMap<>();
+        ArgsRegistry.registerInstabugReportTypesArgs(args);
+        final String[] keysArray = args.keySet().toArray(new String[0]);
+
+        ReadableArray givenArray = new ReadableArray() {
```

Comment:
Do we really need this? There's no way to use the actual object rather than a mock?

Reply:
ReadableArray is a react native type, and it's an abstract class so there's no way to create an instance from it. Would a better approach be to create a class that extends it for testing purposes?